### DISCUSSION TITLE
System updates

### DISCUSSION THREAD

**Discussion by palexdev** [tiann/KernelSU#2301] [Q&A]
Can someone clarify this section of the documentation?

> It is more convenient to upgrade and OTA; when upgrading KernelSU, you can directly install it in the manager without flashing manually; after the system OTA, you can directly install it to the second slot without manual flashing.

After the system OTA I can't install anything as I already lost root at that point. This sentence makes no sense to me? Can I make KernelSU survive system updates and what are the passages

**Comment by WHATHBER** [comment]
It means you should click ![Screenshot_2025-01-02-22-25-01-353_me.weishu.kernelsu-edit.jpg](https://github.com/user-attachments/assets/18f90046-15b5-49ee-a0c7-bf54a406afde)

BEFORE your OTA rebooting, and then you don't need to flash init_boot.img again to stay rooted.
